app/src/behaviors/behavior_sticky_key.c

```diff
@@ -266,7 +303,12 @@ void behavior_sticky_key_timer_handler(struct k_work *item) {
     if (sticky_key->timer_cancelled) {
         sticky_key->timer_cancelled = false;
     } else {
-        release_sticky_key_behavior(sticky_key, sticky_key->release_at);
+        // If the key is lazy, a release is not needed on timeout
+        if (sticky_key->config->lazy) {
+            clear_sticky_key(sticky_key);
+        } else {
+            release_sticky_key_behavior(sticky_key, sticky_key->release_at);
+        }
```

Comment:
This same conditional logic is added in two different place. Can we pull it out into a dedicated function, or move the logic directly into `release_sticky_key_behavior`?

Reply:
I'll pull it out. The logic doesn't really make sense in `release_sticky_key_behavior`, because the logic reads as "if we've timed out, but we are lazy, then we don't need to release the behavior because we never pressed it. Otherwise, release the behavior."